docs/csharp/programming-guide/classes-and-structs/local-functions.md

```diff
@@ -37,7 +37,29 @@ Local functions make the intent of your code clear. Anyone reading you code can
 A local function is defined as a nested method inside a containing member. Its definition has the following syntax:
 
 ```antlr
-<return-value> <method-name> <parameter-list>
+declaration-statement
+   : local-variable declaration ';'
+   | local-constant-declaration ';'
+   | local-function-declaration
+   ;
+
+local-function-declaration
+   : local-function-header local-function-body
+   ;
+
+local-function-header
+   : local-function-modifiers? return-type identifier type-parameter-list?
+        ( formal-parameter-list? ) type-parameter-constraints-clauses
+   ;
+
+local-function-modifiers
+   : (async | unsafe)
+   ;
+
+local-function-body
+   : block
+   | arrow-expression-body
+   ;
```

Comment:
This grammar says that a local function can be `async` or `unsafe`, but not both, which is not true: `async unsafe Task f() { }` compiles fine as a local function (even though you can't use `await` in it).

I would write the grammar like this (it's consistent with [how the spec defines modifiers for methods](https://github.com/dotnet/csharplang/blob/251cbe0/spec/classes.md#methods)):

```antlr
declaration_statement
   : local_variable_declaration ';'
   | local_constant_declaration ';'
   | local_function_declaration
   ;

local_function_declaration
   : local_function_header local_function_body
   ;

local_function_header
   : local_function_modifier* return_type identifier type_parameter_list?
        '(' formal_parameter_list? ')' type_parameter_constraints_clauses
   ;

local_function_modifier
   : 'async'
   | 'unsafe'
   ;

local_function_body
   : block
   | arrow_expression_body
   ;
```

I have also fixed few formal issues with the grammar (missing single quotes, underscores instead of hyphens).

Or should I take this to the Roslyn repo and report back here once these changes are accepted there?